File: apps/macos/MailShieldAgent/Models/ScanResult.swift
import Foundation

struct ScanResultsResponse: Decodable {
    let items: [EmailScanResult]
}

struct EmailScanResult: Decodable, Identifiable, Equatable {
    let id: String
    let gmailMessageId: String
    let threadId: String
    let subject: String
    let sender: String
    let receivedAt: Date
    let scannedAt: Date
    let riskLevel: RiskLevel
    let riskScore: Int
    let summary: String
    let keyReasons: [String]?
    let recommendedAction: String?
    let agentSteps: [PersistedAgentStep]?
    let checks: [AgentCheck]
}

struct PersistedAgentStep: Decodable, Equatable {
    let id: String
    let agentName: String
    let status: String
    let summary: String
}

enum RiskLevel: String, Decodable {
    case low
    case medium
    case high
    case critical
}

struct AgentCheck: Decodable, Identifiable, Equatable {
    let id: String
    let agentName: String
    let title: String
    let status: AgentCheckStatus
    let reason: String
    let evidence: String?
}

enum AgentCheckStatus: String, Decodable {
    case passed
    case warning
    case failed
}

File: apps/core/src/models/scanResult.ts
export type RiskLevel = "low" | "medium" | "high" | "critical";

export type AgentCheckStatus = "passed" | "warning" | "failed";

export type AgentCheck = {
  id: string;
  agentName: string;
  title: string;
  status: AgentCheckStatus;
  reason: string;
  evidence?: string;
};

export type EmailScanResult = {
  id: string;
  gmailMessageId: string;
  threadId: string;
  subject: string;
  sender: string;
  receivedAt: string;
  scannedAt: string;
  riskLevel: RiskLevel;
  riskScore: number;
  summary: string;
  keyReasons?: string[];
  recommendedAction?: string;
  agentSteps?: Array<{
    id: string;
    agentName: string;
    status: "completed" | "limited";
    summary: string;
  }>;
  checks: AgentCheck[];
};

export type ScanResultsResponse = {
  items: EmailScanResult[];
};
